lib/rhc/commands/ssh.rb

```diff
@@ -46,7 +46,11 @@ def run(_, command)
         command_line = [RHC::Helpers.split_path(ssh), ('-vv' if debug?), rest_app.ssh_string.to_s, command].flatten.compact
 
         debug "Invoking Kernel.exec with #{command_line.inspect}"
-        Kernel.send(:exec, *command_line)
+        begin
+          Kernel.send(:exec, *command_line)
+        rescue Errno::EACCES => e
+          raise RHC::PermissionDeniedException.new("Permission denied: do not have permission to access #{ssh}")
```

Comment:
Why not instead fix the checks in `check_ssh_executable!`? It seems like the real defect is in that method because it assumes that the path is fine if the path is a file *or* executable, and we really need it to be a file *and* executable.